source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -111,6 +126,55 @@ MaxmindDbPtr GeoipProvider::initMaxMindDb(const absl::optional<std::string>& db_
   }
 }
 
+CountryMappingPtr
+GeoipProvider::initCountryMapping(const absl::optional<std::string>& country_mapping_path) {
+  auto country_mapping = std::make_unique<absl::flat_hash_map<std::string, std::string>>();
+  if (country_mapping_path) {
+    if (file_system_.fileExists(country_mapping_path.value())) {
+      auto file_or_error = file_system_.fileReadToEnd(country_mapping_path.value());
+
+      if (!file_or_error.status().ok()) {
+        throwEnvoyExceptionOrPanic(fmt::format("Unable to open country mapping file {}. Error {}",
+                                               country_mapping_path.value(),
+                                               std::string(file_or_error.status().message())));
+      }
+      parseCountryMapping(file_or_error.value(), country_mapping_path.value(), country_mapping);
+    } else {
+      config_->incCountryMappingFileParseError();
```

Comment:
practically, this is not file parse error, but rather file does not exist/could not be resolved by filesystem object.